Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,572 @@
+        m_data[get_idx("thetax_max_"+species_names[i_s])] = thetax_max;
+        m_data[get_idx("thetax_std_"+species_names[i_s])] = thetax_std;
+#elif defined(WARPX_DIM_3D)
+        // w_tot
+        // x_ave, y_ave,
+        // thetax_min, thetax_ave, thetax_max
+        // thetay_min, thetay_ave, thetay_max
+        amrex::ReduceOps<ReduceOpSum,
+                         ReduceOpSum, ReduceOpSum,
+                         ReduceOpMin, ReduceOpSum, ReduceOpMax,
+                         ReduceOpMin, ReduceOpSum, ReduceOpMax> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<Real,
+                                                         Real, Real,
+                                                         Real, Real, Real,
+                                                         Real, Real, Real>>(
+            myspc,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<Real,
+                                                                             Real, Real,
+                                                                             Real, Real, Real,
+                                                                             Real, Real, Real>
+            {
+                const amrex::Real w  = p.rdata(PIdx::w);
+                const amrex::Real x = p.pos(0);
+                const amrex::Real y = p.pos(1);
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uy = p.rdata(PIdx::uy);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                const amrex::Real thetay = std::atan2(uy, uz);
+                return {w, w*x, w*y,
+                        thetax, w*thetax, thetax,
+                        thetay, w*thetay, thetay};
+            },
+            reduce_ops);
+
+        amrex::Real w_tot  = amrex::get<0>(r);
+        amrex::Real x_ave  = amrex::get<1>(r);
+        amrex::Real y_ave  = amrex::get<2>(r);
+        amrex::Real thetax_min  = amrex::get<3>(r);
+        amrex::Real thetax_ave  = amrex::get<4>(r);
+        amrex::Real thetax_max  = amrex::get<5>(r);
+        amrex::Real thetay_min  = amrex::get<6>(r);
+        amrex::Real thetay_ave  = amrex::get<7>(r);
+        amrex::Real thetay_max  = amrex::get<8>(r);
+
+        amrex::ParallelDescriptor::ReduceRealSum({w_tot, x_ave, y_ave,
+                                                  thetax_ave, thetay_ave},
+                                                  amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMin({thetax_min, thetay_min}, amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMax({thetax_max, thetay_max}, amrex::ParallelDescriptor::IOProcessorNumber());
+
+        if (w_tot > std::numeric_limits<Real>::min())
+        {
+            x_ave = x_ave / w_tot;
```

Comment:
Perhaps move this block of code into the `if` block below so there is only one check of `w_tot`. This also removes the need for the `else` block.